jupyter_client/ioloop/restarter.py

```diff
@@ -73,8 +85,8 @@ async def poll(self):
                 await self.kernel_manager.restart_kernel(now=True, newports=newports)
                 self._restarting = True
         else:
-            if self._initial_startup:
+            if self._initial_startup and self._last_dead - now >= self.stable_start_time:
```

Comment:
Just as with `KernelManager.shutdown_wait_time` I think we should allow the provisioner to have a say as to what this value should be and use this configured value as a _recommended_ value.  We'd then just extend [`KernelProvisionerBase` with a `get_stable_start_time()` method](https://github.com/jupyter/jupyter_client/blob/f453b51eeeff9e905c583b7da3905c0e35cfbdf0/jupyter_client/provisioning/provisioner_base.py#L207).

Reply:
I've pushed a commit adding this logic. Let me know if I understood you correctly 👍

Reply:
Thanks @vidartf - the change looks correct.